cibuildwheel/util.py

```diff
@@ -574,23 +574,25 @@ def virtualenv(
 T = TypeVar("T", bound=PurePath)
 
 
-def find_compatible_abi3_wheel(wheels: Sequence[T], identifier: str) -> Optional[T]:
+def find_compatible_wheel(wheels: Sequence[T], identifier: str) -> Optional[T]:
     """
-    Finds an ABI3 wheel in `wheels` compatible with the Python interpreter
+    Finds a wheel with an abi3 or a none ABI tag in `wheels` compatible with the Python interpreter
     specified by `identifier`.
     """
 
     interpreter, platform = identifier.split("-")
-    if not interpreter.startswith("cp3"):
-        return None
     for wheel in wheels:
         _, _, _, tags = parse_wheel_filename(wheel.name)
         for tag in tags:
-            if tag.abi != "abi3":
-                continue
-            if not tag.interpreter.startswith("cp3"):
+            if tag.abi == "abi3":
+                if not (interpreter.startswith("cp3") and tag.interpreter.startswith("cp3")):
+                    continue
+            elif tag.abi == "none":
+                if tag.interpreter[:3] != "py3" and tag.interpreter[:3] != interpreter[:3]:
```

Comment:
This line is confusing me... if the wheel tag doesn't start with 'py3', then we check that the part after that is different?

Also, (genuine question), why do we care about python interpreter version if abi is 'none'? To me, 'none' would mean that we don't actually use the Python C interface (e.g. a ctypes/cffi library) I suppose the obvious answer is "that's what `packaging` does", but I'm curious about the semantics of 'none'.

Reply:
I was sure cp35-none would be compatible with cp36, 37, ....
`pip debug` proves me wrong (or there's a bug in packaging).
I will restrict this to `py3`